gr-digital/lib/crc32_bb_impl.cc

```diff
@@ -65,21 +65,40 @@ namespace gr {
       }
     }
 
+    unsigned int
+    crc32_bb_impl::calculate_crc32(const unsigned char* in, size_t packet_length){
+      unsigned int crc = 0;
+      d_crc_impl.reset();
+      if (!d_packed){
+        const size_t n_packed_length = 1 + ((packet_length - 1) / 8);
+        unsigned char packed_buffer[n_packed_length];
```

Comment:
The gods of C99 do smile upon you, while C++'s olymp frowns and shakes its heads:

This isn't valid C++. You'll need to `new` the hell out of this buffer, and free it at the end, or use a `std::vector`. Both would also inherently `memset` it to 0.

Oh, by the way, don't do that. Memory allocation is expensive, and you really don't need to reallocate every time `work` is called. Just give the class a member buffer of some fixed length (a page?), allocate that in the ctor, deallocate in dtor, and call `d_crc_impl.process_bytes()` repeatedly.

If you're in a hurry, go the `std::vector` route. It's ugly, and that's why we do it all over the place.

Reply:
hm, yeah we could limit the unpacked operation to multiple of 8 items. then the call to process bytes on a single byte might work.